### PR TITLE
feat: expose missing process APIs in sandboxed renderers

### PR DESCRIPTION
**Pull request by miniak**
Add the following APIs:

`process.getCPUUsage()`
`process.getIOCounters()`
`process.arch`
`process.pid`
`process.resourcesPath`
`process.sandboxed`
`process.type`
`process.version`
`process.versions`

##### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

Notes: Several missing `process` object APIs exposed to sandboxed renderers

### CODE REVIEW

**Comment by codebytere** [comment]
hi @miniak would you please moving forward include the checklist from the template instead of deleting it as well as write slightly more descriptive PR descriptions for reviewers?

**Comment by miniak** [comment]
@codebytere sorry for that, I've added the checklist.

**Comment by miniak** [comment]
@zeke these props are not new, they are actually defines by node. But only a aubset of the process api is available in sandboxed renderers

**Comment by zeke** [comment]
> these props are not new

Ok let me rephrase: newly exposed

>  only a subset of the process api is available in sandboxed renderers.

Can also you clarify this in the `process`API doc?

**Comment by miniak** [comment]
@zeke I've added the documentation and refactored the implementation a bit

**Comment by miniak** [comment]
I would suggest merging the fix https://github.com/electron/electron/pull/13839 first

**Comment by MarshallOfSound** [comment]
@miniak This now has a bunch of conflicts (I'm assuming with your other PR's)

Once those are cleaned up I'll take a look 👍

**Comment by miniak** [comment]
@MarshallOfSound resolved

File: docs/api/process.md [hunk 1]
```diff
@@ -68,6 +77,11 @@ instead of the `--no-deprecation` command line flag.
 
 A `String` representing the path to the resources directory.
 
+### `process.sandboxed`
+
+A `Boolean`. When the renderer process is sandboxed, this property is `true`,
+otherwise it is `undefined`.
```
Comment (alexeykuzmin) [inline docs/api/process.md]: How difficult is it to make it to be equal to `false` in a non-sandboxes renderer?

Comment (miniak) [reply docs/api/process.md]: @alexeykuzmin it's simple of course. I did it like that as some other properties follow similar pattern (only being set to true if applicable, missing otherwise)

**Comment by alexeykuzmin** [comment]
@miniak Can you please resolve merge conflicts and make sure all the builds are green?

**Comment by miniak** [comment]
@alexeykuzmin conflicts resolved
